File: Go/ExampleTests/common/adder_string.go
package common

type AdderString struct {
	Number1 string
	Number2 string
	Result string
}

func NewAdderStringFromSlice(v []string) AdderString {
	s := AdderString{}
	if len(v) > 0 { s.Number1 = v[0] }
	if len(v) > 1 { s.Number2 = v[1] }
	if len(v) > 2 { s.Result = v[2] }
	return s
}

// NewAdderStringFromText builds from the text form, e.g. Money as "25 USD".
func NewAdderStringFromText(text string) AdderString {
	parts := RequireTokens(text, 3, "Adder")
	return AdderString{
		Number1: parts[0],
		Number2: parts[1],
		Result: parts[2],
	}
}

func (s AdderString) String() string {
	return Token(s.Number1) + " " + Token(s.Number2) + " " + Token(s.Result)
}

func (s AdderString) Equals(o AdderString) bool {
	return DNCEqual(s.Number1, o.Number1) &&
		DNCEqual(s.Number2, o.Number2) &&
		DNCEqual(s.Result, o.Result)
}

func EqualAdderStringSlices(a, b []AdderString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/adder_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct AdderString {
    pub number1: String,
    pub number2: String,
    pub result: String,
}

impl AdderString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            number1: v.get(0).copied().unwrap_or("").to_string(),
            number2: v.get(1).copied().unwrap_or("").to_string(),
            result: v.get(2).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 3, "Adder");
        Self {
            number1: parts[0].clone(),
            number2: parts[1].clone(),
            result: parts[2].clone(),
        }
    }
}

impl std::fmt::Display for AdderString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.number1), crate::common::tokens::token(&self.number2), crate::common::tokens::token(&self.result)].join(" "))
    }
}

impl PartialEq for AdderString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.number1, &other.number1)
            && crate::common::dnc_equal(&self.number2, &other.number2)
            && crate::common::dnc_equal(&self.result, &other.result)
    }
}
